plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitSource.java

```diff
@@ -111,6 +115,7 @@ public HudiSplitSource(
         this.dynamicFilter = requireNonNull(dynamicFilter, "dynamicFilter is null");
         this.dynamicFilteringWaitTimeoutMillis = dynamicFilteringWaitTimeoutMillis.toMillis();
         this.dynamicFilterWaitStopwatch = Stopwatch.createStarted();
+        this.cachingHostAddressProvider = requireNonNull(cachingHostAddressProvider, "cachingHostAddressProvider is null");
```

Comment:
How is the `cachingHostAddressProvider` used? Not very familiar with the caching architecture, are all cached logic handled by the lower levels / parents of the (child) classes?